runtime/Cpp/runtime/src/atn/DecisionState.h

```diff
@@ -12,6 +12,18 @@ namespace atn {
 
   class ANTLR4CPP_PUBLIC DecisionState : public ATNState {
   public:
+    static bool is(const ATNState &atnState) {
+      const auto stateType = atnState.getStateType();
+      return stateType == ATNStateType::BLOCK_START ||
```

Comment:
mmh, so with these many terms comparing with the proposed bitset might be cheaper.

Reply:
The set of enums are technically 2 contiguous spans. So we can shift this to 4 comparisions.

Reply:
but wouldn't a bit-set be easier to read anyway ?

Reply:
We also have requirements on some of the type enums being the same as the serialization values. So using a bitset won't really work unless we implement yet another mapping. Keeping the base classes out of the enum is preferred. Now DecisionState::is and BlockStartState::is are smaller, as it takes advantage of some of the enums being continuous.